src/Microsoft.Windows.CsWin32.Shared/Generator.cs

```diff
@@ -289,16 +289,16 @@ public class Generator : IDisposable
         /// <param name="apiDocsPath">The path to the API docs file.</param>
         /// <param name="options">Options that influence the result of generation.</param>
         /// <param name="compilation">The compilation that the generated code will be added to.</param>
-        /// <param name="parseOptions">The parse options that will be used for the generated code.</param>
-        public Generator(string metadataLibraryPath, string? apiDocsPath, GeneratorOptions? options = null, CSharpCompilation? compilation = null, CSharpParseOptions? parseOptions = null)
+        /// <param name="languageVersion">The language version that will be used for the generated code.</param>
+        public Generator(string metadataLibraryPath, string? apiDocsPath, GeneratorOptions? options = null, CSharpCompilation? compilation = null, LanguageVersion? languageVersion = null)
```

Comment:
📝 This pull request contains a few distinct examples of taking a value that frequently changes and extracting an intermediate incremental value which doesn't change as frequently. Breaking up `compilation` in the same manner could be a high-value win, but it's also significantly more work.

Reply:
The general approach for breaking up `compilation` would be:

* Keep a map of source document &rarr; types of interest declared in that source document
* Keep a map of metadata reference &rarr; types of interest declared in that metadata reference
* Reference the maps in the code generation step instead of directly using `Compilation`

This approach means a change to one source file only requires a change to one value in the maps, and then a quick lookup during the incremental output step.

Reply:
I like the idea. I'm puzzled by your removal of `ParseOptions` though as that should change infrequently. And isn't that where I'd find things like whether NRT is enabled? I don't use that yet but I may someday.

Reply:
> I don't use that yet but I may someday.

None of this is public API. If/when you need something new, feel free to add it. Until then, omitting it means the source generator won't trigger on changes to that item.